Makefile

```diff
@@ -1187,6 +1189,24 @@ tags0:
 format:
 	build_tools/format-diff.sh
 
+check-format:
+	build_tools/format-diff.sh -c
+
+buck-targets:
+	python buckifier/buckify_rocksdb.py
+
+check-buck-targets: buck-targets
+	$(eval TMP=$(shell bash -c "git diff TARGETS | head -n 1"))
```

Comment:
Two issues here:
* You can "do the right thing" (re-build buck target appropriately) and still get an error message if you are running "make check" before commit. I think that should be fine.
  * It might be nice to have "make check" tell you about uncommitted changes, but that should be handled independently
* Supposing someone made intentional updates to TARGETS that should instead go in targets_cfg.py or similar, this check will overwrite them (no backup).

To solve both, I suggest no "git diff" but first backup TARGETS to TARGETS~, then regenerate TARGETS, then diff TARGETS TARGETS~.

Reply:
> Two issues here:
> 
> * You can "do the right thing" (re-build buck target appropriately) and still get an error message if you are running "make check" before commit. I think that should be fine.
>   

Good point, and this needs to be fixed by only running `check-buck-targets` if TARGETS file does not have uncommitted change. I didn't consider this since this was originally for Github Action.

>   * It might be nice to have "make check" tell you about uncommitted changes, but that should be handled independently
> * Supposing someone made intentional updates to TARGETS that should instead go in targets_cfg.py or similar, this check will overwrite them (no backup).

Also agree. Will make a copy of TARGETS.
> 
> To solve both, I suggest no "git diff" but first backup TARGETS to TARGETS~, then regenerate TARGETS, then diff TARGETS TARGETS~.